editor/src/core/shared/javascript-cache.ts

```diff
@@ -8,9 +8,11 @@ import { SafeFunctionCurriedErrorHandler } from './code-exec-utils'
 
 type JavaScriptContainer = JSXAttributeOtherJavaScript | ArbitraryJSBlock | JSXArbitraryBlock
 
-export type GetOrUpdateFunctionCache = (javascript: JavaScriptContainer) => Function
+export type GetOrUpdateFunctionCache = (
+  javascript: JavaScriptContainer,
+) => (...args: Array<any>) => any
 
-let functionCache: { [uniqueID: string]: Function } = {}
+let functionCache: { [uniqueID: string]: (...args: Array<any>) => any } = {}
```

Comment:
```suggestion
let functionCache: { [uniqueID: string]: () => any } = {}
```

I _think_ we can get away  using this shorter notation

Reply:
That wouldn't work because then we wouldn't be able to call the functions from the cache with arguments inside `resolveParamsAndRunJsCode`. The only alternative would be try replacing the `any`s with `unknowns` (though it looks like you'd have to make further changes elsewhere for that - I haven't followed it all the way through I just tested that really quickly in this one file)

Reply:
then it is fine like this, `unknown` is only useful on the "inside" of a function, not when talking about the function's shape as it is passed around

Reply:
Well, I strongly disagree, since that's how we have ended up with implicit casts to `any` that have caused us many issues throughout, but I don't think it's an immediate issue here anyway.

Reply:
is there `any` reason not to go with `(...args: Array<unknown>) => unknown` here, and further down? have made the change because it seems better, but i really don't get this bit of code.

Reply:
i ended up with `any` here because it didn't like `unknown`, but maybe someone wants to take a look at this one because the function cache could maybe be a bit more strongly typed

Reply:
I think this is fine as it stands, because this pretty much mirrors what we had previously anyway.